### PR TITLE
add startmode and boot protocol object

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

**Comment by mchf** [comment]
hm seems to be a bit overkill and netconfig centric ... but i have to read it in more depth

**Comment by jreidinger** [comment]
@mchf well, I discuss it also with knut how it looks like on NM and there is also something like startmode, but they support different values ( only autoconnect or not ). So in future I expect that we need also to have a flag for each class what backend it supports.

File: src/lib/y2network/interface_config_builder.rb [hunk 1]
```diff
@@ -129,6 +133,34 @@ def firewall_zone=(value)
...
+    end
+
+    def startmode=(name)
+      mode = Startmode.create(name)
+      # assign only if it is not already this value. This helps with ordering of ifplugd_priority
+      @connection_config.startmode = mode if @connection_config.startmode.name != mode.name
+      @config["STARTMODE"] = mode.name
+    end
+
+    def ifplugd_priority=(value)
```
Comment (imobachgs) [inline src/lib/y2network/interface_config_builder.rb]: Please, document (at least the @param). Same for `startmode=` and `ifplugd_priority`

File: src/lib/y2network/startmodes/auto.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/startmode"
+
+module Y2Network
+  module Startmodes
+    class Auto < Startmode
```
Comment (imobachgs) [inline src/lib/y2network/startmodes/auto.rb]: Please, document. I know that for us the meaning is clear, but I am thinking about newcomers. And the same for all startmode classes.

**Comment by imobachgs** [comment]
Oh, wait, you need to solve the conflicts first.

**Pull request merged**